scripts/releases/prebuild/build.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ * @oncall react_native
+ */
+
+const {HEADERS_FOLDER, TARGET_FOLDER} = require('./constants');
+const {getAllFilesRecursively} = require('./folders');
+const {execSync} = require('child_process');
+const fs = require('fs');
+const path = require('path');
+
+/*::
+import type { Dependency, Platform } from './types';
+*/
+
+/**
+ * Builds dependencies for the specified platforms. This function will use the generated
+ * Package.swift file together with the extracted dependencies to build the frameworks for
+ * the requested platforms.
+ */
+async function buildDepenencies(
+  scheme /*: string */,
+  dependencies /*: $ReadOnlyArray<Dependency> */,
+  platforms /*: $ReadOnlyArray<Platform> */,
+  rootFolder /*: string */,
+  buildFolder /*: string */,
+) {
+  console.log('✅ Building dependencies...');
+
+  await Promise.all(
+    platforms.map(platform =>
+      buildPlatform(scheme, platform, rootFolder, buildFolder),
+    ),
+  );
+
+  // Copy headers into framework
+  await copyHeadersToFrameworks(scheme, dependencies, rootFolder, buildFolder);
+}
+
+/**
+ * Builds a single platform.
+ */
+async function buildPlatform(
+  scheme /*: string */,
+  platform /*: Platform */,
+  rootFolder /*: string */,
+  buildFolder /*: string */,
+) {
+  console.log(`Building ${platform}...`);
+  const command =
+    `xcodebuild -scheme "${scheme}" -destination "generic/platform=${platform}" ` +
+    `-derivedDataPath "${buildFolder}" ` +
+    '-configuration Release ' +
```

Comment:
we will only building for Release, in this way.
Shouldn't we build for both?

Reply:
XCFramworks cannot contain both release/debug build of the same arch/platform, even when using fat binaries unfortunately- so I'm building in release mode WITH dSYMs :) That should be a starting point until we find out how we'd like it to end up.